### PR TITLE
CavcDetail storybook

### PR DESCRIPTION
**Pull request by hschallhorn**
### Description
Get storybook component for CavcDetail up and running
Referenced:
1. https://storybook.js.org/docs/react/essentials/controls
2. https://github.com/department-of-veterans-affairs/caseflow/blob/3784761f15723e3d1a775db3a46be7022494a8dc/client/app/components/TextField.stories.mdx

### Testing Plan
1. Run storybook locally. This should automatically open a tab for storybook
```bash
make storybook
```
(I got errors and needed to follow [these](https://dsva.slack.com/archives/CQNPUG8EQ/p1599058657001500?thread_ts=1599058592.001400&cid=CQNPUG8EQ) instructions)
2. Go to Queue -> Components -> CavcDetail
3. Edit the controls to see what happens!

### Storybook Story
*For Frontend (Presentationa) Components*
* [x] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [x] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component (we sould do this with default values for jmr vs mdr)

Screen Shot 2020-10-30 at 12 07 00 PM

### CODE REVIEW

File: client/app/queue/CaseDetailsView.jsx [hunk 1]
```diff
@@ -143,7 +143,7 @@ export const CaseDetailsView = (props) => {
             <AppellantDetail title="About the Appellant" appeal={appeal} />
           )}
           {!_.isNull(appeal.cavc_remand) && appeal.cavc_remand &&
-          (<CavcDetail title="CAVC Remand" appealId={appealId} />)}
+          (<CavcDetail title="CAVC Remand" {...appeal.cavc_remand} />)}
```
Comment (hschallhorn) [inline client/app/queue/CaseDetailsView.jsx:146]: Makes this a presentational component, allowing us to easily create a storybook entry for this

File: client/app/queue/CavcDetail.jsx [hunk 2]
```diff
@@ -90,12 +85,14 @@ const CavcDetail = ({ appeal }) => {
...
+  decision_date: PropTypes.string.isRequired,
+  judgement_date: PropTypes.string,
+  mandate_date: PropTypes.string
 };
 
-const mapStateToProps = (state, ownProps) => ({
-  appeal: appealWithDetailSelector(state, { appealId: ownProps.appealId })
-});
-
-export default connect(mapStateToProps)(CavcDetail);
```
Comment (hschallhorn) [inline client/app/queue/CavcDetail.jsx:101]: Props are directly passed, no need to go fetch the appeal out of state

Comment (ajspotts) [reply client/app/queue/CavcDetail.jsx:101]: awesome, great adjustment thank you!

**Review by ajspotts — APPROVED** [review]
Thank you for taking the time to make the changes and do up the PR, really appreciate it!!!

**Pull request merged**